core/org.wso2.carbon.base/src/test/java/org/wso2/carbon/base/CarbonBaseUtilsTest.java

```diff
@@ -0,0 +1,124 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Test class for CarbonBaseUtil related methods with a correct sec policy.
+ */
+public class CarbonBaseUtilsTest {
+
+    @BeforeClass
+    public void setSecurityManager() {
+        URL resourceURL = CarbonBaseUtilsTest.class.getClassLoader().getResource("");
+        if (resourceURL != null) {
+            String resourcePath = resourceURL.getPath();
+            resourcePath = resourcePath + "policy-test.policy";
+            System.setProperty("java.security.policy", resourcePath);
+            System.setSecurityManager(new SecurityManager());
+        }
+    }
+
+    @AfterClass
+    public void clearSecurityManager() {
+        System.clearProperty("java.security.policy");
+    }
+
+    @Test(groups = {"org.wso2.carbon.base"})
+    public void testGetCarbonConfigDirPath() throws ClassNotFoundException, IllegalAccessException,
+            NoSuchFieldException {
+        //setting env variables for testing getCarbonConfigDirPath() in CarbonBaseUtils
+        Map<String, String> newEnv = new HashMap<>();
+        newEnv.put(CarbonBaseConstants.CARBON_CONFIG_DIR_PATH_ENV, "../wso2am-2.1.0/repository/conf");
+        try {
+            Class<?> processEnvironmentClass = Class.forName("java.lang.ProcessEnvironment");
+            Field theEnvironmentField = processEnvironmentClass.getDeclaredField("theEnvironment");
+            theEnvironmentField.setAccessible(true);
+            Map<String, String> env = (Map<String, String>) theEnvironmentField.get(null);
+            env.putAll(newEnv);
+            Field theCaseInsensitiveEnvironmentField = processEnvironmentClass.getDeclaredField
+                    ("theCaseInsensitiveEnvironment");
+            theCaseInsensitiveEnvironmentField.setAccessible(true);
+            Map<String, String> cienv = (Map<String, String>) theCaseInsensitiveEnvironmentField.get(null);
+            cienv.putAll(newEnv);
+        } catch (NoSuchFieldException e) {
+            Class[] classes = Collections.class.getDeclaredClasses();
+            Map<String, String> env = System.getenv();
+            for (Class cl : classes) {
+                if ("java.util.Collections$UnmodifiableMap".equals(cl.getName())) {
+                    Field field = null;
+                    field = cl.getDeclaredField("m");
+                    field.setAccessible(true);
+                    Object obj = field.get(env);
+                    Map<String, String> map = (Map<String, String>) obj;
+                    map.clear();
+                    map.putAll(newEnv);
+                }
+            }
+        }
+        //setting env variables done for testing getCarbonConfigDirPath() in CarbonBaseUtils
+        assertEquals(CarbonBaseUtils.getCarbonConfigDirPath(),
+                "../wso2am-2.1.0/repository/conf", "Must provide the location where carbon.xml resides");
```

Comment:
It would make better sense if we write this message as "Carbon config directory location not set as an environment variable" since it focuses more on the root cause. Carbon config path necessarily does not mean carbon.xml path.